tests/_serialization/test_simple_properties.py

```diff
@@ -240,13 +240,13 @@ def test_linkorelse_deserialize_requires_serializable():
 
 def test_linkorelse_deserialize_requires_scope_and_id():
     loe = LinkOrElse()
-    with pytest.raises(ValueError):
+    with pytest.raises(TypeError):
```

Comment:
This appears to be an interface break. If so, we need to either force it to a `ValueError` for the time being and make the switch come 3.0, or make sure we very clearly and proactively (read: not just release notes) communicate when it happens, so people can update their scripts if need be.

Reply:
Ironically, solvable with the try-except block that @bfolie asked me to replace with the signature test, albeit in this case the exception in thrown by a dependency.  Thinking about how to address this, given the harder issue below, but it's worth noting that this case should really be a TypeError because missing __init__ arguments are TypeErrors in the Python core libraries.

Reply:
It makes sense to me this should be a `TypeError`, but that means a user's `try-except` won't catch it any more. So as annoying as it is, I do think we should catch it, and make a note to change it at the 3.0 release.